MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java

```diff
@@ -0,0 +1,290 @@
+    setContentView(R.layout.activity_animated_pin_drop);
+
+    // Initialize the map view
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+  }
+
+  @Override
+  public void onMapReady(@NonNull final MapboxMap mapboxMap) {
+    mapboxMap.setStyle(Style.LIGHT, new Style.OnStyleLoaded() {
+      @Override
+      public void onStyleLoaded(@NonNull Style style) {
+        ValueAnimatorIconAnimationActivity.this.style = style;
+        mapView.addOnDidFinishRenderingMapListener(ValueAnimatorIconAnimationActivity.this);
+        initLayerIcon(style);
+        initDataSource(style);
+      }
+    });
+  }
+
+  /**
+   * Implementing this interface so that animation only starts once all tiles have been loaded
+   *
+   * @param fully whether or not the mapy is finished rendering
+   */
+  @Override
+  public void onDidFinishRenderingMap(boolean fully) {
+    initAnimation(currentSelectedTimeInterpolator);
+    initInterpolatorButtons();
+  }
+
+  /**
+   * Add images to the map so that the SymbolLayers can reference the images.
+   */
+  private void initLayerIcon(@NonNull Style loadedMapStyle) {
+    loadedMapStyle.addImage(ICON_ID, BitmapFactory.decodeResource(
+      getResources(), R.drawable.map_marker_push_pin_pink));
+  }
+
+  /**
+   * Add GeoJsonSource with random Features to the map.
+   */
+  private void initDataSource(@NonNull Style loadedMapStyle) {
+    // Add a new source from the GeoJSON data
+    loadedMapStyle.addSource(
+      new GeoJsonSource("source-id",
+        FeatureCollection.fromFeatures(new Feature[] {
+          Feature.fromGeometry(Point.fromLngLat(
+            119.86083984375,
+            -1.834403324493515)),
+          Feature.fromGeometry(Point.fromLngLat(
+            116.06637239456177,
+            5.970619502704659)),
+          Feature.fromGeometry(Point.fromLngLat(
+            114.58740234375,
+            4.54357027937176)),
+          Feature.fromGeometry(Point.fromLngLat(
+            118.19091796875,
+            5.134714634014467)),
+          Feature.fromGeometry(Point.fromLngLat(
+            110.36865234374999,
+            1.4500404973608074)),
+          Feature.fromGeometry(Point.fromLngLat(
+            109.40185546874999,
+            0.3076157096439005)),
+          Feature.fromGeometry(Point.fromLngLat(
+            115.79589843749999,
+            1.5159363834516861)),
+          Feature.fromGeometry(Point.fromLngLat(
+            113.291015625,
+            -0.9667509997666298)),
+          Feature.fromGeometry(Point.fromLngLat(
+            116.40083312988281,
+            -0.3392008994314591))
+        })
+      )
+    );
+  }
+
+  /**
+   * Initialize and start the animation.
+   *
+   * @param desiredTimeInterpolator the type of Android system movement to animate the
+   *                                SymbolLayer icons with.
+   */
+  private void initAnimation(TimeInterpolator desiredTimeInterpolator) {
```

Comment:
need to check if animator is not null, if yes call cancel on it

Reply:
Fixed